OutOfSchool/OutOfSchool.WebApi/Controllers/ChatWorkshopController.cs

```diff
@@ -56,30 +56,179 @@ public class ChatWorkshopController : ControllerBase
         }
 
         /// <summary>
-        /// Get a chat room with information about Parent and Workshop.
+        /// Get parent's chat room with information about Parent and Workshop.
         /// </summary>
         /// <param name="id">ChatRoom's Id.</param>
         /// <returns>User's chat room that was found.</returns>
-        [HttpGet("chatrooms/{id}")]
+        [HttpGet("parent/chatrooms/{id}")]
+        [Authorize(Roles = "parent")]
         [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ChatRoomWorkshopDto))]
         [ProducesResponseType(StatusCodes.Status204NoContent)]
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
         [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-        public async Task<IActionResult> GetRoomByIdAsync([Range(1, long.MaxValue)] long id)
+        public Task<IActionResult> GetRoomForParentByRoomIdAsync([Range(1, long.MaxValue)] long id)
+            => this.GetRoomByIdAsync(id, chatRoom => this.ParentHasRigtsForChatRoomAsync(chatRoom.ParentId));
```

Comment:
1. This lambda `chatRoom => this.ParentHasRigtsForChatRoomAsync(chatRoom.ParentId)` and it's twin is heavily duplicated. Why not introduce helper method that can be reused?
2. `ParentHasRigtsForChatRoomAsync` - please, fix the mistype - it should be `...Rights..`

Reply:
thank you very much!
Done.
I changed the original method to take not Id but ChatRoom.

Reply:
Cool, this is even better